crates/typst-library/src/visualize/image/svg.rs

```diff
@@ -26,12 +28,36 @@ struct Repr {
     tree: usvg::Tree,
 }
 
+fn svg_parse_inner(
+    data: &Bytes,
+    options: &usvg::Options,
+    span: Span,
+    file: Option<FileId>,
+) -> SourceResult<usvg::Tree> {
+    if data.starts_with(&[0x1f, 0x8b]) {
```

Comment:
Good point, I'll add that, this is indeed needed to do proper error handling which is just... unfortunate :( Because for binary SVGs, we don't have text (to generate an error) until it's been decompressed